tls/s2n_ktls.h

```diff
@@ -42,7 +42,16 @@ typedef int (*s2n_setsockopt_fn)(int socket, int level, int option_name,
 S2N_RESULT s2n_ktls_set_setsockopt_cb(s2n_setsockopt_fn cb);
 
 bool s2n_ktls_is_supported_on_platform();
+S2N_RESULT s2n_ktls_get_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
 
 /* These functions will be part of the public API. */
 int s2n_connection_ktls_enable_send(struct s2n_connection *conn);
 int s2n_connection_ktls_enable_recv(struct s2n_connection *conn);
+
+/* Testing */
+/* msghdr is not `const` for sendmsg because the real implementation will need to modify
+ * ancillary data on msghdr prior to sending */
+typedef ssize_t (*s2n_ktls_sendmsg_fn)(struct s2n_connection *conn, struct msghdr *msg, uint8_t record_type);
+typedef ssize_t (*s2n_ktls_recvmsg_fn)(struct s2n_connection *conn, struct msghdr *msg, uint8_t *record_type);
+S2N_RESULT s2n_ktls_set_send_recv_msg_fn(s2n_ktls_sendmsg_fn send_cb, s2n_ktls_recvmsg_fn recv_cb);
```

Comment:
Yea, good question. Since these are set as a global value (Note how it doesnt take a connection arg) and only used for testing I wanted a convenient and quick way to set them for both the server and client in one-shot.

Having two fn might make things more verbose. What do you think?

Reply:
It's not super important, but it's kind of more "s2ny" to treat send and recv as being configured separately. I think I would probably change this to:

Reply:
Yea i originally had it like that and wasnt sure. Ill change it back